### ISSUE TITLE
Planar hex produces ParaView incompatable mesh

### ISSUE THREAD

**Issue by pwolfram** [MPAS-Dev/MPAS-Tools#262]
The following command produces a culled_mesh.nc that cannot be opened in paraview:

```
planar_hex --nx 6 --ny 28 --dc 1000.0 --npy -o grid.nc
MpasCellCuller.x grid.nc culled_mesh.nc                       
```

The error message in ParaView is 

```
ERROR: In /Users/kitware/dashboards/buildbot-slave/8275bd07/build/superbuild/paraview/src/VTK/IO/NetCDF/vtkMPASReader.cxx, line 1539
vtkMPASReader (0x7ffc04418090): Exceeded storage for extra points!


ERROR: In /Users/kitware/dashboards/buildbot-slave/8275bd07/build/superbuild/paraview/src/VTK/Common/ExecutionModel/vtkExecutive.cxx, line 784
vtkPVCompositeDataPipeline (0x7ffc04422440): Algorithm vtkMPASReader(0x7ffc04418090) returned failure for request: vtkInformation (0x7ffc02ae4b90)
  Debug: Off
  Modified Time: 1010209
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_DATA
  FROM_OUTPUT_PORT: 0
  ALGORITHM_AFTER_FORWARD: 1
  FORWARD_DIRECTION: 0

```

My assumption is that somehow we can produce a mesh that works in the MPAS codes but is incompatible with the specification used to produce the ParaView reader.

**Comment by pwolfram** [comment]
cc @xylar, have you seen this issue (or can you at least reproduce this bug)?

**Comment by xylar** [comment]
I can reproduce this using paraview 5.5.0.  I believe it's likely to be a ParaView issue rather than a `planar_hex` issue:
https://gitlab.kitware.com/paraview/paraview/issues/18036
It could be that the same error is coming up for other reasons, but I believe ParaView was never very good at handling periodic meshes.

**Comment by xylar** [comment]
I attempted to create the same conditions with `periodic_hex` to try to eliminate the possibility that `planar_hex` introduced this problem.

With the default settings (500x500 10-km mesh), ParaView is able to open both the output from `periodic_hex` and an identically sized mesh from `planar_hex` (after I run it through the mesh converter or do some other step to add the `Time` dimension that ParaVeiw expects for some reason).  However, I created your 6x30 1-km mesh (ny = 30 before culling in the y direction) in both tools and the result is that ParaVeiw is not able to open either one.  I therefore thing this has nothing to do with culling and only to do with the size of the meshes.

I don't think we can address this here, at least not without first getting help from KitWare to track down the problem.  I therefore suggest you open an issue with them, and provide an example mesh that isn't working to them (or give them the steps to create it).  If you agree, please close this issue here once you've opened an issue with them.

**Comment by pwolfram** [comment]
@xylar, given that this is an outstanding issue I would argue we should leave this open until it is fixed.  It isn't 100% clear that this is their problem-- it may actually be a bug with `planar_hex`.  

To go further, are we aware of or have any `planar_hexes` that open currently in ParaView?  If so, it would be good to share this mesh with them too.  We could also have a bug in `MpasCellCuller.x` too for periodic meshes.  Are we 100% sure this isn't the case?  

Given we don't know precisely the problem we need to keep this issue open so that the bug isn't lost-- this way next time we have an opportunity to have KitWare work on something we can give them a clean list of bugs (and this would be one of them).

**Comment by demarle** [comment]
There is a fix for this problem in ParaView 5.6.0. Can you try that version (or later) and let us know if the problem persists?

**Comment by pwolfram** [comment]
@demarle, will try with [ParaView-5.6.1-MPI-OSX10.12-64bit.unsigned.dmg](https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v5.6&type=binary&os=macOS&downloadFile=ParaView-5.6.1-MPI-OSX10.12-64bit.unsigned.dmg) and report back-- thanks!

**Comment by pwolfram** [comment]
@demarle and @xylar -- it is fixed via the new version 5.6.1 and the mesh opens!   Thank you!

**Comment by xylar** [comment]
Thanks @demarle!
